NXP imx943 add gpt as system tick

yongxu-wang15 opened the pull request:
Added gpt node for imx943 Mcore, use GPT1 for M33, GPT2 for M70, GPT3 for M71, aimed to support system suspend by it

yongxu-wang15 committed:
boards: nxp: imx943: enable GPT hardware timers for system tick

Enable GPT hardware timers on imx943_evk boards:
- gpt_hw_timer1 for M33 core
- gpt_hw_timer2 for M7_0 core
- gpt_hw_timer3 for M7_1 core

These will be used as system tick sources when PM is enabled.

Signed-off-by: Yongxu Wang <yongxu.wang@nxp.com>

yongxu-wang15 committed:
soc: nxp: imx943: configure GPT as system timer for power management

Configure GPT timers as system tick source when PM is enabled to resolve
low power mode issues:
- Cortex-M SysTick loses state during mix power-off
- SysTick cannot maintain time during system suspend

Configuration changes:
- Disable Cortex-M SysTick when GPT timer is enabled
- Set GPT clock frequency to 32768 Hz for low power operation, 32k is
  reserved in soc level system sleep mode
- Configure system tick rate to 1024 Hz

Signed-off-by: Yongxu Wang <yongxu.wang@nxp.com>

yongxu-wang15 committed:
dts: arm: imx943: add GPT timer nodes for system tick and counter

Add GPT timer nodes to support system tick and counter functionality:

- gpt_hw_timer1/2/3: System timers for M33/M7_0/M7_1 cores
- gpt4: General purpose timer/counter

Signed-off-by: Yongxu Wang <yongxu.wang@nxp.com>

kylebonnici commented:
You can work around this by running linter locally fix files locally until https://github.com/zephyrproject-rtos/zephyr/pull/98187 its merged ... for this PR I have uploaded the diff

 
[diff.patch](https://github.com/user-attachments/files/23162333/diff.patch)

kylebonnici commented:
@yongxu-wang15  PR is merged rebase and try again

JiafeiPan commented on soc/nxp/imx/imx9/imx943/Kconfig.defconfig.mimx94398.m7_0:
```diff
@@ -17,8 +17,12 @@ config MCUX_CORE_SUFFIX
 config SYS_CLOCK_HW_CYCLES_PER_SEC
-	default $(dt_node_int_prop_int,/cpus/cpu@0,clock-frequency)
+	default $(dt_node_int_prop_int,/cpus/cpu@0,clock-frequency) if CORTEX_M_SYSTICK
+	default 32768 if MCUX_GPT_TIMER
```
is this frequency "32768"  a fixed value?

yongxu-wang15 replied:
yes, as the gpt system timer driver directly use 32KHz clock frequency which can reserved to work in LP mode in mostly soc, so its a fixed value now.

JiafeiPan commented:
And fix compliance issue.

yongxu-wang15 committed:
dts: arm: nxp: fix imx943 compliance ci check formatted warning

Fix Device Tree formatting issues to comply with Zephyr coding style
guidelines for i.MX943 EVK board and dts files

Signed-off-by: Yongxu Wang <yongxu.wang@nxp.com>
